src/pgen/cluster/agn_triggering.cpp

```diff
@@ -437,20 +466,34 @@ AGNTriggeringReduceTriggering(parthenon::MeshData<parthenon::Real> *md,
     break;
   }
   }
+  if (agn_triggering.triggering_mode_ == AGNTriggeringMode::COLD_GAS) {
+    auto &agn_triggering = hydro_pkg->Param<AGNTriggering>("agn_triggering"); // Make it non-const
+    const parthenon::Real cold_mass = hydro_pkg->Param<Real>("agn_triggering_cold_mass");
+    const parthenon::Real total_mass = hydro_pkg->Param<Real>("agn_triggering_total_mass");
+
+    hydro_pkg->UpdateParam<Real>("agn_triggering_prev_cold_mass", cold_mass); //Move from AGNTriggeringFinalizeTriggering
+    hydro_pkg->UpdateParam<Real>("agn_triggering_prev_total_mass", total_mass);
+  }
+  hydro_pkg->UpdateParam<AGNTriggering>("agn_triggering", agn_triggering);
   return TaskStatus::complete;
 }
 
 parthenon::TaskStatus
 AGNTriggeringMPIReduceTriggering(parthenon::StateDescriptor *hydro_pkg) {
 #ifdef MPI_PARALLEL
-  const auto &agn_triggering = hydro_pkg->Param<AGNTriggering>("agn_triggering");
+  auto &agn_triggering = hydro_pkg->Param<AGNTriggering>("agn_triggering");
   switch (agn_triggering.triggering_mode_) {
   case AGNTriggeringMode::COLD_GAS: {
 
     Real accretion_rate = hydro_pkg->Param<Real>("agn_triggering_cold_mass");
     PARTHENON_MPI_CHECK(MPI_Allreduce(MPI_IN_PLACE, &accretion_rate, 1,
                                       MPI_PARTHENON_REAL, MPI_SUM, MPI_COMM_WORLD));
     hydro_pkg->UpdateParam("agn_triggering_cold_mass", accretion_rate);
+
+    Real total_mass = hydro_pkg->Param<Real>("agn_triggering_total_mass");
+    PARTHENON_MPI_CHECK(MPI_Allreduce(MPI_IN_PLACE, &total_mass, 1,
+                                          MPI_PARTHENON_REAL, MPI_SUM, MPI_COMM_WORLD));
+    hydro_pkg->UpdateParam("agn_triggering_total_mass", total_mass);
```

Comment:
Given that `MPI_Allreduce` are expensive operations, I suggest to combine both numbers, c.f., https://github.com/parthenon-hpc-lab/athenapk/blob/f8497c5f39473c557b84c7078e1eb696246e139c/src/pgen/turbulence.cpp#L384